src/main/java/net/jonathangiles/tools/teenyhttpd/model/TypedResponse.java

```diff
@@ -0,0 +1,38 @@
+package net.jonathangiles.tools.teenyhttpd.model;
+
+import net.jonathangiles.tools.teenyhttpd.implementation.TypedResponseImpl;
+
+public interface TypedResponse<T> extends Response {
+
+    T getValue();
+
+    TypedResponse<T> header(String key, String value);
+
+    static <T> TypedResponse<T> ok() {
```

Comment:
Can we use the same pattern as Response for these static factory methods and call them create? Also, we might want to consider adding asTypedResponse to the StatusCode enum.

Reply:
I think, `create` looks quite redundant sometimes 😅 for example

```java
import static net.jonathangiles.tools.teenyhttpd.model.TypedResponse.create;
import static net.jonathangiles.tools.teenyhttpd.model.StatusCode.*;

return create(StatusCode.OK, new Pet(name, age, type))
```

Yes but

```java
return create(CREATED, new Pet(name, age, type))
```


**comparative**

```java
@Get("/product/:id")
    public TypedResponse<Product> getProduct(@PathParam("id") int id) {

        for (Product product : productList) {
            if (product.getId() == id) {
                return ok(product);
            }
        }

        return status(NOT_FOUND);
    }
```


```java
@Get("/product/:id")
    public TypedResponse<Product> getProduct(@PathParam("id") int id) {

        for (Product product : productList) {
            if (product.getId() == id) {
                return create(OK, product);
            }
        }

        return create(NOT_FOUND);
    }
```

--- 

I'll add asTypedResponse to the StatusCode enum. StatusCode is already handled by the `apply` method in `EndpointHandler` We can simply write something like this:

```java
 @Get("/statusCode")
 public StatusCode returnStatusCode() {
     return FORBIDDEN;
 }
``` 

> EndpointHandler will handle this as `new EmptyResponse((StatusCode) result)`

Reply:
If you want to improve the Response API for the common status code scenarios, that is fine too!